packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -282,7 +300,7 @@ class HotRunner extends ResidentRunner {
   }
 
   Future<bool> _updateDevFS({ bool fullRestart = false }) async {
-    if (!_refreshDartDependencies()) {
+    if (!await _refreshDartDependencies()) {
```

Comment:
Do we want to run pub if dirty on reload, or just on restart? Only on restart feels slightly more natural to me, as its handling a bigger change.

Reply:
Gotcha. We could also message on reload (`You pubspec is out of date, perform a hot restart to get new dependencies`), but users would then just need to perform the restart. So, this change looks good for both reload and restart.